### PR TITLE
Add bottom subnav

### PR DESCRIPTION
**Pull request by nicl**
(Required for MVP.)

Subnav has been refactored to be more self-contained as part of this change, to enable easier reuse.

screen shot 2018-09-27 at 10 35 38

### CODE REVIEW

File: frontend/components/Header/Nav/SubNav.tsx [hunk 1]
```diff
@@ -111,8 +118,10 @@ const parentStyle = css`
 `;
 
 interface Props {
-    parent?: LinkType;
-    links: LinkType[];
+    subnav?: {
+        parent?: LinkType | undefined;
```
Comment (SiAdcock) [inline frontend/components/Header/Nav/SubNav.tsx]: Do we need to specify this as a union type if it's an optional field? Asking for a friend 👀

File: frontend/components/Header/Nav/SubNav.tsx [hunk 2]
```diff
@@ -164,8 +173,22 @@ export default class Subnav extends Component<
     }
 
     public render() {
-        const { parent, links } = this.props;
+        const sn = this.props.subnav;
+
+        if (sn && sn.links) {
```
Comment (SiAdcock) [inline frontend/components/Header/Nav/SubNav.tsx]: `sn.links` is mandatory, so maybe this check is superfluous?

File: frontend/components/Header/Nav/SubNav.tsx (hunk 1 above)
Comment (nicl) [reply frontend/components/Header/Nav/SubNav.tsx]: It's a good point actually. I just copied the original type definition, but I'm not sure it adds anything here.

**Comment by nicl** [comment]
@SiAdcock Hey does this look good now? Thanks for comments.

**Comment by nicl** [comment]
@SiAdcock thanks man!

**Pull request merged**